src/hooks/usePokemonTypes/index.js

```diff
@@ -0,0 +1,20 @@
+import {useState, useEffect, useCallback} from 'react';
+import axios from "axios";
+
+export default function usePokemonTypes(pokemonApiUrl) {
+    const [types, setTypes] = useState([]);
+
+    const getAllPokemonTypes = useCallback( async () => {
+        const typesList =  await axios.get(pokemonApiUrl + '/type');
+        const typeName = typesList.results.map(type => type.name);
+        setTypes(typeName);
+    }, [pokemonApiUrl])
+
+    useEffect(() => {
+        if (types && types.length === 0) {
```

Comment:
You can leave just types.length and do not need here === 0, because if statement always convert variables to boolean type.